11-Backbonejs/Backbone-Models.md

```diff
@@ -0,0 +1,309 @@
+
+For example:
+
+```javascript
+// src/app.js
+import $ from 'jquery';
+import _ from 'underscore';
+
+import Task from 'app/models/task';
+
+var my_task = new Task({
+  title: "Create a model",
+  completed: true
+});
+  // prints "Title:  Create a model"
+console.log("Title: " + my_task.get("title") );
+  // prints "Complete:  true"
+console.log("Complete: " + my_task.get("completed") );
+```
+
+The `set` method allows you to change the attributes of a Model by either passing it a key and value or by passing `set` a JSON object.
+
+```javascript
+import $ from 'jquery';
+import _ from 'underscore';
+
+import Task from 'app/models/task';
+
+var my_task = new Task({
+  title: "Create a model",
+  completed: true
+});
+
+  // key & value example
+my_task.set("title", "Test the `set` method.");
+  // prints "Title:  Test the `set` method."
+console.log("Title: " + my_task.get("title") );
+
+  // using an object
+my_task.set({title: "Run a hash of object"});
+  // prints "Title:  Create a model"
+console.log("Title: " + my_task.get("title") );
+  // prints "Complete:  true"
+console.log("Complete: " + my_task.get("completed") );
+```
+
+That's... great, but lets get something output to the webpage.
+
+### Creating An Underscore Template
+
+Lets use our fabulous underscore library to create a template to render, or draw our task item.  
+
+```html
+<!-- build/index.html -->
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="utf-8">
+</head>
+<body>
+  <!-- Where the main content of our page will go -->
+  <main>
+  </main>
+  <div id="test-area">
+  </div>
+  <script id="taskItemTemplate" type="text/template">
+    <section class="task-item">
+      <h1><strong>Title:</strong> <%= title %></h1>
+      <h3><strong>Completed:</strong> <%= completed %></h3>
+    </section>
+  </script>
+
+  <script src="/app.bundle.js" charset="utf-8"></script>
+</body>
+</html>
+
+```
+
+Note inside the `body` element is an underscore template.  Next we'll use the template to add content.  
+
+To use the jQuery template first we select the template text using jQuery.  Next we use the `_.template` method to get a template object by passing it the text of the template block.  The `<%= %>` handles tell underscore to insert a value in from the parameters passed to the object when it's called.  Lastly we append the compiled template to the contents of the `main` element using the fields in `my_task`.  
+
+```javascript
+// src/app.js
+import $ from 'jquery';
+import _ from 'underscore';
+
+import Task from 'app/models/task';
+
+var my_task = new Task({
+  title: "Create a model",
+  completed: true
+});
+
+// Added Code
+  // Select the template using jQuery
+var template_text = $('#taskItemTemplate').html();
+  // Get an underscore template object
+var template = _.template(template_text);
+  // 
+$('main').append(template(my_task.toJSON()));
```

Comment:
We could manually append text to the file...  Since Dan had had a previous section on using Underscore templates, I figured we'd start with that.